element_array_ephys/export/nwb.py

```diff
@@ -0,0 +1,153 @@
+import datajoint as dj
+import numpy as np
+import json
+import pynwb
+
+from element_array_ephys import ephys as ephys_acute
+from element_array_ephys import ephys_chronic, ephys_no_curation, probe
+
+
+def curated_clusterings_to_nwb(curated_clustering_keys, nwbfile, ephys_module=None):
+    """
+    Generate one NWBFile object representing all ephys data
+     coming from the specified "curated_clustering_keys"
+    Note: the specified "curated_clustering_keys" must be all from one session
+
+    :param curated_clustering_keys: entries of CuratedClustering table
+    :param nwbfile: nwbfile object containing session meta information
+    :param ephys_module: the activated ephys module calling this function
+    :return: NWBFile object
+    """
+    if ephys_module is None:
+        for ephys_module in (ephys_acute, ephys_chronic, ephys_no_curation):
+            if ephys_module.schema.is_activated():
+                ephys = ephys_module
+                break
+        else:
+            raise ValueError(f'No activated ephys module found!')
+    else:
+        ephys = ephys_module
+
+    # validate input
+    if isinstance(curated_clustering_keys, dj.expression.QueryExpression):
+        curated_clustering_keys = (ephys.CuratedClustering & curated_clustering_keys).fetch('KEY')
+
+    assert len(ephys._linking_module.Session & curated_clustering_keys) == 1, \
+        f'Multiple sessions error! The specified "curated_clustering_keys"' \
+        f' must be from one session only'
+
+    # create processing module for LFP
+    ecephys_module = nwbfile.create_processing_module(name='ecephys',
+                                                      description='preprocessed ephys data')
+
+    # add additional columns to the electrodes table
+    electrodes_query = probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode
+    for additional_attribute in ['shank', 'shank_col', 'shank_row']:
+        nwbfile.add_electrode_column(
+            name=electrodes_query.heading.attributes[additional_attribute].name,
+            description=electrodes_query.heading.attributes[additional_attribute].comment)
+
+    # add additional columns to the units table
+    units_query = ephys.EphysRecording * ephys.ClusteringTask @ ephys.CuratedClustering.Unit
+    for additional_attribute in ['cluster_quality_label', 'spike_count', 'sampling_rate',
+                                 'spike_times', 'spike_sites', 'spike_depths']:
```

Comment:
Getting the following error regarding spike times:
>UserWarning: Column 'spike_times' is predefined in Units with index=True which does not match the entered index argument. The predefined index spec will be ignored. Please ensure the new column complies with the spec. This will raise an error in a future version of HDMF.
  warn(msg)

Reply:
You don't need to add 'spike_times', it is by default part of the Units table. Just omit "spike_times" this from the `add_unit_column` calls and keep it in the `add_unit` calls